src/classes/HiFiMixerSession.ts

```diff
@@ -186,16 +188,25 @@ export class HiFiMixerSession {
     onUsersDisconnected: Function;
     /**
      * This function is called when the "connection state" changes.
-     * Right now, this is called when the the RAVI session state changes to
-     * `RaviSessionStates.CONNECTED`, `RaviSessionStates.DISCONNECTED`, and `RaviSessionStates.FAILED`.
      */
-    onConnectionStateChanged: Function;
+    onConnectionStateChanged: ConnectionStateChangeHandler;
 
     /**
      * Contains information about the mixer to which we are currently connected.
      */
     mixerInfo: any;
+    
+    /**
+     * Track whether or not we're in the process of attempting to connect. (This is
+     * a little bit hacky, but given the maze of twisty callbacks and Promises, it
+     * seems the most sane way to avoid double-connecting.)
+     */
+    _tryingToConnect: boolean;
```

Comment:
I lost track: in the end, did it turn out that we always go to state Connecting when we're trying to connect? (In which case why do we need this as well?)

Or maybe that Connecting means we're generally connecting, retrying, waiting for a retry, etc., while this is really a micro state means that we're not in a waiting between attempts, but have specifically initiated an attempt?  (In which case, let's include a comment to that effect?)

Reply:
The latter -- Connecting and Reconnecting mean that we're generally "in the greater process of trying to connect" and this is a micro state. I will add a comment!